diff --git a/index.d.ts b/index.d.ts
index 567431591f995485f37ccd67b2756ec837b45a9b..04699c5c3add9b1a4f706131b898ccd7af6f5f43 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -395,6 +395,7 @@
   | Primitive
   | {toJSON: () => string}
   | ReadonlyArray<SerializableParam>
+  | ReadonlySet<SerializableParam>
   | Readonly<{[key: string]: SerializableParam}>;

 interface AtomFamilyOptionsWithoutDefault<T, P extends SerializableParam> {
