diff --git a/index.d.ts b/index.d.ts
index 8fb62646440fd73bff9d2709d37c8b18640f43e7..b79cda56389c2e3c4e88fcd582b27cb7980a12be 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -258,7 +258,7 @@ export function isRecoilValue(val: unknown): val is RecoilValue<any>; // eslint-
 // bigint not supported yet
 type Primitive = undefined | null | boolean | number | symbol | string;
 
-export type SerializableParam = Primitive | ReadonlyArray<SerializableParam> | Readonly<{[key: string]: SerializableParam}>;
+export type SerializableParam = Primitive | ReadonlyArray<SerializableParam> | ReadonlySet<SerializableParam> | Readonly<{[key: string]: SerializableParam}>;
 
 export interface AtomFamilyOptions<T, P extends SerializableParam> {
   key: NodeKey;
