diff --git a/dist/cjs/server/framework-preset-react.js b/dist/cjs/server/framework-preset-react.js
index 44a44514a2a545e7203ff9643f6bcb6178cc780a..e2fea0171e2b553d22de49f19d9aff031295d53f 100644
--- a/dist/cjs/server/framework-preset-react.js
+++ b/dist/cjs/server/framework-preset-react.js
@@ -45,6 +45,13 @@ var context = storybookReactDirName.includes('node_modules') ? _path.default.joi
 
 var hasJsxRuntime = function () {
   try {
+    // https://github.com/storybookjs/storybook/pull/13899
+    try {
+      require.resolve('react', { paths: [context] })
+    } catch (e) {
+      require.resolve('react/jsx-runtime');
+      return true;
+    }
     require.resolve('react/jsx-runtime', {
       paths: [context]
     });
diff --git a/dist/esm/server/framework-preset-react.js b/dist/esm/server/framework-preset-react.js
index 208639651e8d1f77319d5aed789b34e6fe36beed..1b7b93ee919c97193034d2cd5624b8198d8d92d4 100644
--- a/dist/esm/server/framework-preset-react.js
+++ b/dist/esm/server/framework-preset-react.js
@@ -29,6 +29,13 @@ var context = storybookReactDirName.includes('node_modules') ? path.join(storybo
 
 var hasJsxRuntime = function () {
   try {
+    // https://github.com/storybookjs/storybook/pull/13899
+    try {
+      require.resolve('react', { paths: [context] })
+    } catch (e) {
+      require.resolve('react/jsx-runtime');
+      return true;
+    }
     require.resolve('react/jsx-runtime', {
       paths: [context]
     });
