diff --git a/src/components/Input.d.ts b/src/components/Input.d.ts
index eb5c7027b379591a23b857e62bf23be057c3ab69..5b4cf0f16fdfe079f5387983a913415175c23065 100755
--- a/src/components/Input.d.ts
+++ b/src/components/Input.d.ts
@@ -1,16 +1,16 @@
-import { ComponentType, ReactElement, Ref as ElementRef } from 'react';
+import { ComponentType, ReactElement, InputHTMLAttributes } from 'react';
 import { CSSObject } from '@emotion/serialize';
 
 import { colors, spacing } from '../theme';
 
 import { ClassNamesState, Theme } from '../types';
 
-export interface InputProps {
+export type InputProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'ref'> & {
     cx: (a: string | null, b: ClassNamesState, c: string) => string | void;
     getStyles: (name: string, props: any) => {};
     theme: Theme;
     /** Reference to the internal element */
-    innerRef: (element: ElementRef<any>) => void;
+    innerRef: (element: HTMLInputElement | null) => void;
     /** Set whether the input should be visible. Does not affect input size. */
     isHidden: boolean;
     /** Whether the input is disabled */
